client/history/client.go

```diff
@@ -282,7 +282,7 @@ func (c *clientImpl) DeleteDLQTasks(
 }
 
 func (c *clientImpl) getAnyClient(apiName string) (historyservice.HistoryServiceClient, error) {
```

Comment:
I didn't see when this went in, but do you think it would be so bad to do `c.redirector.clientForShardID(randomShardId())`? yeah, it's not round-robin, but for low traffic is fine, and it's much simpler code.

It seems better to have less code messing with the connection cache directly.

Alternatively, what about the `membership://history` idea? I think one concern is it would create a second grpc connection to each history node, but maybe that's okay.

Reply:
> I didn't see when this went in, but do you think it would be so bad to do c.redirector.clientForShardID(randomShardId())? yeah, it's not round-robin, but for low traffic is fine, and it's much simpler code.
> 
> It seems better to have less code messing with the connection cache directly.

I agree that this does seem better--not a strong enough reason to break this abstraction. However, I'm still going to use a deterministic shard ID to make testing deterministic without requiring DI'ing a rand.Source.

> Alternatively, what about the membership://history idea? I think one concern is it would create a second grpc connection to each history node, but maybe that's okay.

Can you elaborate on this? I actually forget what it was.

Reply:
it would be a new method on RPCFactory: CreateAnyHistoryGRPCConnection, which would basically boil down to Dialing the string `membership://history`, which will create a single grpc.ClientConn that round-robins among history members under the hood. then something would have to wrap that in a HistoryClient with retries and metrics.

I don't think it's worth it here, but in some ways it's cleaner than doing it manually like this, because you can use grpc configuration to control the load balancing and stuff.